scripts/migration/GPII-4014/verify.js

```diff
@@ -0,0 +1,163 @@
+
+// Usage: node scripts/migration/GPII-4014/verify.js CouchDB-url clientCredentialId-for-NOVA ...
+// @param {String} CouchDB-url - The url to the CouchDB where docoments should be verified.
+// @param {Strings} clientCredentialIds-for-NOVA - The "_id" value of the NOVA client credential. There could be any
+// number of client credential parameters from here onwards.
+
+// A sample command that runs this script in the universal root directory:
+// node scripts/migration/GPII-4014/verify.js http://localhost:25984 "clientCredential-nova1" "clientCredential-nova2"
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    url = require("url");
+
+fluid.registerNamespace("gpii.migration.GPII4014");
+
+require("./shared/migratedValues.js");
+require("../../shared/dbRequestUtils.js");
+require("../../../gpii/node_modules/gpii-db-operation/src/DbUtils.js");
+
+/**
+ * Create a set of options for this script.
+ * The options are based on the command line parameters and a set of database constants.
+ * @param {Array} processArgv - The command line arguments.
+ * @return {Object} - The options.
+ */
+gpii.migration.GPII4014.initOptions = function (processArgv) {
+    var options = {};
+    options.couchDbUrl = processArgv[2] + "/gpii";
+    options.novaClientCredentials = process.argv.splice(3);
+
+    // Set up database specific options
+    options.allDocsUrl = options.couchDbUrl + "/_all_docs?include_docs=true";
+    options.allDocs = [];
+    options.parsedCouchDbUrl = url.parse(options.couchDbUrl);
+    options.postOptions = {
+        hostname: options.parsedCouchDbUrl.hostname,
+        port: options.parsedCouchDbUrl.port,
+        path: "/gpii/_bulk_docs",
+        auth: options.parsedCouchDbUrl.auth,
+        method: "POST",
+        headers: {
+            "Accept": "application/json",
+            "Content-Length": 0,          // IMPORTANT: FILL IN PER REQUEST
+            "Content-Type": "application/json"
+        }
+    };
+    console.log("COUCHDB_URL: '" +
+        options.parsedCouchDbUrl.protocol + "//" +
+        options.parsedCouchDbUrl.hostname + ":" +
+        options.parsedCouchDbUrl.port +
+        options.parsedCouchDbUrl.pathname + "'"
+    );
+    return options;
+};
+
+/**
+ * Create the step that retrieves all documents from the database
+ * @param {Object} options - All docs URL and whether to filter:
+ * @param {Array} options.allDocsUrl - The url for retrieving all documents in the database.
+ * @return {Promise} - A promise that resolves retrieved documents.
+ */
+gpii.migration.GPII4014.verifyAllDocs = function (options) {
+    var details = {
+        requestUrl: options.allDocsUrl,
+        requestErrMsg: "Error retrieving documents from the database: ",
+        responseDataHandler: gpii.migration.GPII4014.verifyDocsData,
+        responseErrMsg: "Error retrieving documents from database: "
+    };
+    return gpii.dbRequest.configureStep(details, options);
```

Comment:
It seems like we should improve the configureStep utility so that the responseDataHandler is allowed to return a promise indicating whether the processing has been successful or not. Otherwise we will only actually end with an error status in the case that the DB operation itself fails, regardless of whether the response passes verification